pkg/controller/sync/dispatch/managed.go

```diff
@@ -299,15 +315,23 @@ func (d *managedDispatcherImpl) setResourcesUpdated() {
 	d.resourcesUpdated = true
 }
 
-func (d *managedDispatcherImpl) CollectedStatus() status.CollectedPropagationStatus {
+func (d *managedDispatcherImpl) CollectedStatus() (status.CollectedPropagationStatus, status.CollectedResourceStatus) {
 	d.RLock()
 	defer d.RUnlock()
 	statusMap := make(status.PropagationStatusMap)
+	resourceStatusMap := make(map[string]interface{})
 	for key, value := range d.statusMap {
 		statusMap[key] = value
 	}
-	return status.CollectedPropagationStatus{
-		StatusMap:        statusMap,
-		ResourcesUpdated: d.resourcesUpdated,
+
+	for key, value := range d.resourceStatusMap {
+		resourceStatusMap[key] = value
 	}
+	return status.CollectedPropagationStatus{
+			StatusMap:        statusMap,
+			ResourcesUpdated: d.resourcesUpdated,
```

Comment:
should this rather be a field of `CollectedPropagationStatus`?

Reply:
Can you elaborate more the reasoning behind this decision ?

Reply:
well it's a field with the same value. Actually it looks like `ResourcesUpdated` on `CollectedResourceStatus` is never used. Not sure if it's bug or if it's just redundant. Might be redundant because it always has the same value as the other one 🤷